cf/domaintopology.py

```diff
@@ -0,0 +1,235 @@
+    three-dimensional volume cells that correspond to a volume
+    enclosed by a set of faces, but how the nodes relate to volume
+    boundary vertices is undefined and so volume cells are currently
+    omitted from the CF data model.
+
+    See CF Appendix I "The CF Data Model".
+
+    **NetCDF interface**
+
+    The netCDF variable name of the UGRID mesh topology variable may
+    be accessed with the `nc_set_variable`, `nc_get_variable`,
+    `nc_del_variable`, and `nc_has_variable` methods.
+
+    .. versionadded::  3.16.0
+
+    """
+
+    @property
+    def cell(self):
+        """The cell type.
+
+        {{cell type}}
+
+        .. versionadded:: 3.16.0
+
+        .. seealso:: `del_cell`, `get_cell`, `has_cell`, `set_cell`
+
+        """
+        return self.get_cell(default=AttributeError())
+
+    @cell.setter
+    def cell(self, value):
+        self.set_cell(value)
+
+    @cell.deleter
+    def cell(self):
+        self.del_cell(default=AttributeError())
+
+    def identity(
+        self,
+        default="",
+        strict=None,
+        relaxed=False,
+        nc_only=False,
+        relaxed_identity=None,
+    ):
+        """Return the canonical identity.
+
+        By default the identity is the first found of the following:
+
+        * The cell type type, preceded by ``'cell:'``.
+        * The `standard_name` property.
+        * The `id` attribute, preceded by ``'id%'``.
+        * The `long_name` property, preceded by ``'long_name='``.
+        * The netCDF variable name, preceded by ``'ncvar%'``.
+        * The value of the *default* parameter.
+
+        .. versionadded:: 3.16.0
+
+        .. seealso:: `id`, `identities`, `long_name`, `cell`,
+                     `nc_get_variable`, `standard_name`
+
+        :Parameters:
+
+            default: optional
+                If no identity can be found then return the value of the
+                default parameter.
+
+            strict: `bool`, optional
+                If True then the identity is the first found of only
+                the `cell` attribute, "standard_name" property or the
+                "id" attribute.
+
+            relaxed: `bool`, optional
+                If True then the identity is the first found of only
+                the `cell `attribute, the "standard_name" property,
+                the "id" attribute, the "long_name" property or the
+                netCDF variable name.
+
+            nc_only: `bool`, optional
+                If True then only take the identity from the netCDF
+                variable name.
+
+        :Returns:
+
+                The identity.
+
+        """
+        if nc_only:
+            if strict:
+                raise ValueError(
+                    "'strict' and 'nc_only' parameters cannot both be True"
+                )
+
+            if relaxed:
+                raise ValueError(
+                    "'relaxed' and 'nc_only' parameters cannot both be True"
+                )
```

Comment:
Good catch. I opted for 4def36593aeff70ae0ef0960cc0e52937194b2bd in the end. This is because there is more logic in the `if nc_only` branch after the parameter testing, and the values of the parameters might not be `True` (they could be anything that evaluates to True in a Boolean context). We could have moved to a `sum()` approach, but that is slower and this code is also performance critical (can have a has a noticeable impact in `cf.aggregate`)